crash/session.py

```diff
@@ -29,7 +30,42 @@ def __init__(self, kernel_exec=None, vmcore=None, kernelpath=None,
         if not kernel_exec:
             return
 
-        error = gdb.execute("file {}".format(kernel_exec), to_string=True)
+        try:
+            kdump = kdumpfile(vmcore)
+        except OSErrorException as e:
+            raise RuntimeError(str(e))
+
+        kaslr_off = long(
+            kdump.attr.get("linux.vmcoreinfo.lines.KERNELOFFSET", "0"),
+            base=16)
+
+        error = gdb.execute("exec-file {}".format(kernel_exec), to_string=True)
+        tinfo = gdb.execute("info target", to_string=True)
+        args=""
+        textaddr = 0
+        in_exec = False
+        for line in tinfo.splitlines():
```

Comment:
Probably just  a matter of taste, I'd move the parsing out of `__init__`.

Reply:
It looked OK to me for a Hackweek, but if Jeff is also concerned, I can put it in its own method.